### PR TITLE
[CORE-1444] Make DeleteRepos and DeletePipelines similar

### PR DESCRIPTION
**Pull request by robert-uhl**
This PR makes repo & pipeline deletion similar:

- `DeleteRepo` & `DeletePipeline` are intended to delete a single repo or pipeline. `DeletePipeline`’s `all` member is still supported, but deprecated — it only deletes all pipelines in the default project (for backwards compatibility, since the default project is the only one visible to clients ≤ v2.4).
- `pachctl delete {repo,pipeline} --all` delete all repos or pipelines in the current project.
- `pachctl delete {repo,pipeline} --all --all-projects` delete all repos or pipelines in all projects.

### CODE REVIEW

**Review by loganripplinger — COMMENTED** [review]
LGTM

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -2955,6 +2959,46 @@ func (a *apiServer) deletePipelineInTransaction(txnCtx *txncontext.TransactionCo
...
+		if !request.GetAll() && !projects[pipelineInfo.GetPipeline().GetProject().GetName()] {
+			return nil
+		}
+		dr.Pipeline = pipelineInfo.Pipeline
+		err := a.deletePipeline(ctx, dr)
+		if err == nil {
+			deleted[pipelineInfo.Pipeline.String()] = struct{}{}
+			pp = append(pp, pipelineInfo.Pipeline)
+		}
+		return err
```
Comment (albscui) [inline src/server/pps/server/api_server.go:2995]: Maybe aggregate this error into a `multierror` as well?

Comment (robert-uhl) [reply src/server/pps/server/api_server.go:2995]: Ditto here — I think that it is probably better to terminate early rather than continue.

**Review by albscui — APPROVED** [review]
LGTM

**Pull request merged**